control-plane/plugin/src/resources/node.rs

```diff
@@ -432,23 +483,22 @@ impl Drain for Node {
                                 }
                             }
                         }
-                        Err(e) => {
-                            println!("Failed to get node {}. Error {}", id, e);
-                            break;
+                        None => {
+                            println!("Node {} is not registered", id);
                         }
                     }
-                    if timeout_instant.is_some() && time::Instant::now() > timeout_instant.unwrap()
-                    {
-                        println!("Node {} drain command timed out", id);
-                        break;
-                    }
-                    let sleep = Duration::from_secs(1);
-                    tokio::time::sleep(sleep).await;
+                }
+                Err(e) => {
+                    println!("Failed to get node {}. Error {}", id, e);
+                    break;
                 }
             }
-            Err(e) => {
-                println!("Failed to get node {}. Error {}", id, e)
+            if timeout_instant.is_some() && time::Instant::now() > timeout_instant.unwrap() {
+                println!("Node {} drain command timed out", id);
+                break;
             }
+            let sleep = Duration::from_secs(1);
```

Comment:
nit: watch default period is 2 seconds, might want to consider doing the same here?

Reply:
Sure, now fixed.